### PR TITLE
#162949209 Document the API using Swagger

### PR DESCRIPTION
**Pull request by nesh-dev**
#### What does this PR do?
Implement Swagger documentation 
#### Description of Task to be completed?
- install django-rest-swagger
- add swagger configuration in settings.py 
- Add swagger url to the base application url files
#### How should this be manually tested?
- git clone 
- cd into ah-the-unsullied folder 
- git checkout `ft-document-api-with-swagger-162949209`
- run on terminal `pip install -r requirements.txt`
- run migrations `python manage.py makemigrations` then `python manage.py migrate`
- create a super user to run the migrations that is `python manage.py createsuperuser`
- run the app `python manage.py runserver`
- visit the localhost link and test the endpoints on swagger
#### What are the relevant pivotal tracker stories?
 - [#162949209](https://www.pivotaltracker.com/story/show/162949209)
#### Screenshots (if appropriate)

### CODE REVIEW

**Review by kwanj-k — DISMISSED** [review]

**Review by sylviawanjiku — DISMISSED** [review]

**Review by Allan690 — CHANGES_REQUESTED** [review]
Please include a colon before the word "Document" keeping in line with the format outlined[ here](https://github.com/andela/getmyshop/wiki/Commit-Message-Convention#format-of-the-commit-message) 

screenshot 2019-01-10 at 21 41 47

**Review by kwanj-k — APPROVED** [review]

**Review by kwanj-k — CHANGES_REQUESTED** [review]
Make sure the commit message uses lower case only

**Review by Allan690 — DISMISSED** [review]

**Review by Allan690 — DISMISSED** [review]
LGTM

**Review by MbugwaSami — DISMISSED** [review]

**Review by kwanj-k — DISMISSED** [review]

**Review by kelvinwm — DISMISSED** [review]

**Review by indungu — CHANGES_REQUESTED** [review]
The [review app](https://ah-the-unsullied-staging-pr-7.herokuapp.com/) is currently down. This makes it hard to approve this changes since merging them might bring down staging. Please address this ASAP.

**Comment by kwanj-k** [comment]
The review app is now up. 👍

**Review by kwanj-k — DISMISSED** [review]

**Review by MbugwaSami — DISMISSED** [review]

**Review by Allan690 — DISMISSED** [review]

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -1,16 +1,14 @@
 from rest_framework import status
-from rest_framework.generics import RetrieveUpdateAPIView
+from rest_framework import generics
```
Comment (indungu) [inline authors/apps/authentication/views.py]: This import could me more specific, please switch it to one with just the Classes you intend to use.

`from rest_framework.generic import CreateAPIView, RetrieveUpdateAPIView`

File: requirements.txt [hunk 2]
```diff
@@ -4,8 +4,8 @@ django-cors-middleware==1.3.1
 django-extensions==2.1.4
 djangorestframework==3.9.0
 gunicorn==19.9.0
-psycopg2==2.7.6.1
 psycopg2-binary==2.7.6.1
 PyJWT==1.7.1
-psycopg2-binary==2.7.6.1
 whitenoise==4.1.2
+django-rest-swagger==2.1.0
```
Comment (indungu) [inline requirements.txt:14]: There is a newer version of this dependency, please use version `2.2.0`.

File: authors/settings.py [hunk 3]
```diff
@@ -146,9 +147,28 @@
     'EXCEPTION_HANDLER': 'authors.apps.core.exceptions.core_exception_handler',
     'NON_FIELD_ERRORS_KEY': 'error',
 
-    'DEFAULT_AUTHENTICATION_CLASSES': (
-        # 'authors.apps.authentication.backends.JWTAuthentication',
-        'rest_framework.authentication.SessionAuthentication',
-    ),
+    # 'DEFAULT_AUTHENTICATION_CLASSES': (
```
Comment (indungu) [inline authors/settings.py]: Remove all commented out code.

**Review by indungu — CHANGES_REQUESTED** [review]
Please address all the highlighted issues. The endpoints are also not working as expected.
image

File: authors/apps/authentication/views.py (hunk 1 above)
Comment (nesh-dev) [reply authors/apps/authentication/views.py]: The custom JSON renderer has changed the format of the data. It should be passed as included in the README file.

File: requirements.txt (hunk 2 above)
Comment (nesh-dev) [reply requirements.txt:14]: Version 2.2 has issues with input fields, There was a recommendation to donwgrade.

**Review by Allan690 — DISMISSED** [review]

**Review by kwanj-k — DISMISSED** [review]

**Review by Allan690 — APPROVED** [review]

**Review by sylviawanjiku — APPROVED** [review]

**Review by indungu — APPROVED** [review]

**Pull request merged**